teps/0056-pipelines-in-pipelines.md

```diff
@@ -352,6 +365,335 @@ main-`PipelineRun` to be able to fetch and access the sub-`PipelineRun`'s full s
+
+## Alternatives
+
+### Specification - Group `PipelineRef` and `PipelineSpec`
+
+We could add a new type to pass to the `Pipeline` field - say `PipelineTaskPipeline` - which will take `PipelineRef` 
+and `PipelineSpec` only initially and can be expanded to support other fields from [`PipelineRunSpec`][pipelinerunspec].
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipeline:
+        pipelineRef:
+          name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+However, this solution involves duplication that will be worsened over time as we support more features and fields
+from [`PipelineRunSpec`][pipelinerunspec]. It is also inconsistent with the specification for `Tasks` in a `Pipeline`.
+
+### Specification - Use `PipelineRunSpec` in `PipelineTask`
+
+We could add a `PipelineRun` field in `PipelineTask` which would take the [`PipelineRunSpec`][pipelinerunspec]. This 
+field could take `PipelineRef` and `PipelineSpec` only  initially. This provides the extensibility to support other 
+fields related to creating a `PipelineRun` from [`PipelineRunSpec`][pipelinerunspec], beyond the `PipelineRef` and 
+`PipelineSpec` fields only.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipelineRun:
+        pipelineRef:
+          name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+However, this solution mixes up authoring time and run time concerns which is against the Tekton [design principle][dp]
+of *Reusability* through separating authoring time and run time concerns:
+
+> At authoring time (i.e. when authoring `Pipelines` and `Tasks`), authors should be able to include anything that is
+  required for every execution of the `Task` or `Pipeline`. At run time (i.e. when invoking a `Pipeline` or `Task` via
+  `PipelineRun` or `TaskRun`), users should be able to control execution as needed by their context without having to
+  modify `Tasks` and `Pipeline`.
+
+### Specification - Reorganize `PipelineTask`
+
+We could reorganize the `PipelineTask` to better support scaling as we add more types that can be specified and 
+executed in a `Pipeline`, beyond `Tasks` and `Custom Tasks`. We could support the existing `TaskRef` and `TaskSpec`
+fields for the foreseeable future, but anything specified under the new `Ref` or `Spec` fields would take priority. 
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      ref:
+        task:
+          name: git-clone
```

Comment:
Uhm, this seems to simply move the `taskRef` key to `ref.task`.  
I think a neat approach would be use the same syntax as for custom tasks:

```yaml
spec:
  tasks:
    - name: git-clone
      ref:
        apiVersion: tekton.dev/v1beta1
        kind: Task
        name: git-clone
```

We could even have nice defaults for `Task`. Adding in the syntax for `tektoncd/resolution`, the YAML could look like:


```yaml
spec:
  tasks:
    - name: git-clone
      ref:
        name: git-clone
      resolver: (...)
      resource: (...)
```

Reply:
@afrittoli finally got around to opening an issue to track this api change: https://github.com/tektoncd/pipeline/issues/5138 - thank you!